marathon/src/test/scala/io/buoyant/marathon/v2/AppIdNamerTest.scala

```diff
@@ -49,6 +49,24 @@ class AppIdNamerTest extends FunSuite with Awaits {
     assert(state == Activity.Ok(NameTree.Neg))
   }
 
+  test("Namer handles looking up both appId and /appId") {
+    class TestApi() extends Api {
+      def getAppIds(): Future[Api.AppIds] = {
+        Future.value(Set[String]("/foo", "/servicename"))
+      }
+      def getAddrs(app: String): Future[Set[SocketAddress]] = Future.never
+    }
+
+    val namer = new AppIdNamer(new TestApi(), Path.Utf8("io.l5d.marathon"), 250.millis)
+    @volatile var state: Activity.State[NameTree[Name]] = Activity.Pending
+
+    namer.lookup(Path(Buf.Utf8("servicename"), Buf.Utf8("residual"))).states respond { state = _ }
+    assert(state == Activity.Ok(NameTree.Leaf(Path(Buf.Utf8("io.l5d.marathon"), Buf.Utf8("servicename")))))
+
+    namer.lookup(Path(Buf.Utf8("/servicename"), Buf.Utf8("residual"))).states respond { state = _ }
+    assert(state == Activity.Ok(NameTree.Leaf(Path(Buf.Utf8("io.l5d.marathon"), Buf.Utf8("/servicename")))))
+  }
```

Comment:
Per offline discussion, we want hex-encoded paths with slashes, at marathon enforces application IDs prepended with slashes. I've update the PR per the rest of the feedback.